docs/framework-sealights_agent.md

```diff
@@ -0,0 +1,43 @@
+# Sealights Agent Framework
+The Sealights Agent Framework causes an application to be automatically configured to work with [Sealights Service][].
+
+<table>
+  <tr>
+    <td><strong>Detection Criterion</strong></td><td>Existence of a single bound sealights service. The existence of a sealights service defined by the <a href="http://docs.cloudfoundry.org/devguide/deploy-apps/environment-variable.html#VCAP-SERVICES"><code>VCAP_SERVICES</code></a> payload containing a service name, label or tag with <code>sealights</code> as a substring.
+</td>
+  </tr>
+  <tr>
+    <td><strong>Tags</strong></td><td><tt>sealights-agent=&lt;version&gt;</tt></td>
+  </tr>
+</table>
+Tags are printed to standard output by the buildpack detect script
+
+## User-Provided Service
+When binding Sealights using a user-provided service, it must have name or tag with `sealights` in it.
+The credential payload can contain the following entries. 
+
+| Name | Description
+| ---- | -----------
+| `token` | A Sealights Agent token
+
+## Configuration
+For general information on configuring the buildpack, including how to specify configuration values through environment variables, refer to [Configuration and Extension][].
```

Comment:
A couple notes here:

1. For agent-specific settings, like build_session_id & lab_id, if you can link to documentation that explains these, that is helpful to users, especially newer users.
2. There are a couple of conventions around proxy configuration, to make this more automatic for users. First, the buildpack itself respects `http_proxy` and `https_proxy` ([see here](https://github.com/cloudfoundry/java-buildpack/blob/main/docs/extending-repositories.md#proxies)). If these env variables are set, the buildpack automatically uses the defined proxy to download resources. It is very common for users to set these values, when a proxy is required. Second, the JVM has its own proxy settings, `http.proxyHost` and `http.proxyPort`. These are typically set by a user via the `JAVA_OPTS` env variable and would be present when the app runs. Does your agent support any of these existing conventions? If so, it may automatically configure the proxy. If it doesn't, I would suggest that the buildpack look at http_proxy/https_proxy and, if set, automatically configure the proxy settings for your agent. This should remove the step for a user to manually configure proxy settings in a third location, and because it's so common for users that require a proxy to set these already, it'll likely just work for them without any additional thought.
3. You're missing a `version` label. Something like `version` -> `The version of Sealights Agent to use. Candidate versions can be found in [this listing](link-to-your-index.yml-file)` is common.

Reply:
1. Added links to the official docs
2. TL;DR: It's intentional. The http_proxy and https_proxy seem to be mostly relevant for the buildpack phase (please correct me if I'm wrong). This configuration is used at runtime to communicate with the Sealights backend service. It may, or may not, be relevant for all other HTTP operations, therefore we also don't use the http.proxyHost and http.proxyPort system properties -- we've seen a lot of cases where the Sealights' proxy was configured and http.proxyHost wasn't. Additionally, we've seen a few cases where the http.proxyHost/Port were updated at runtime, as part of testing (which is when Sealights agents are used), and wasn't cleared afterwards.
3. Added with the correct links

Reply:
1. & 3. 👍 

For 2., I'm OK with that. If you intentionally want it to not automatically pick up proxy settings. I just know that users often expect stuff like that to just work. What I will do, so it's clear, is add a note to the doc that specifically says that `proxy` is not automatically  deduced from http_proxy/https_proxy or similar JVM settings.